newrelic-agent/src/main/java/com/newrelic/agent/rpm/RPMConnectionServiceImpl.java

```diff
@@ -99,6 +99,14 @@ public long getAppServerPortTimeout() {
         return APP_SERVER_PORT_TIMEOUT;
     }
 
+    /**
+     * Checks whether tasks have begun piling up in the scheduled executor's work queue.
+     * Used to prevent repeated reconnect attempts in the event of a LicenseException.
+     */
+    public boolean shouldPreventNewConnectionTask(){
+        return scheduledExecutor.getQueue().size() > MAX_QUEUED_CONNECT_TASKS;
```

Comment:
If `scheduledExecutor` is switched back to `ScheduledExecutorService` then we will need to cast it to `ScheduledThreadPoolExecutor` to get access to the `getQueue()` method. It does mean that we now are dependent on the `ScheduledThreadPoolExecutor` implementation in this method but doing the explicit casting only in this one place lessens the likely hood that it will spread.

```
    public boolean shouldPreventNewConnectionTask(){
        return ((ScheduledThreadPoolExecutor) scheduledExecutor).getQueue().size() > MAX_QUEUED_CONNECT_TASKS;
    }
```

Reply:
I don't think it matter much but this could also be `>= MAX_QUEUED_CONNECT_TASKS`